File: java/src/Ice/src/main/java/IceInternal/RequestHandler.java
// **********************************************************************
//
// Copyright (c) 2003-2014 ZeroC, Inc. All rights reserved.
//
// This copy of Ice is licensed to you under the terms described in the
// ICE_LICENSE file included in this distribution.
//
// **********************************************************************

package IceInternal;

public interface RequestHandler extends CancellationHandler
{
    RequestHandler connect();
    RequestHandler update(RequestHandler previousHandler, RequestHandler newHandler);

    void prepareBatchRequest(BasicStream out)
        throws RetryException;
    void finishBatchRequest(BasicStream out);
    void abortBatchRequest();

    int sendAsyncRequest(OutgoingAsyncBase out)
        throws RetryException;

    Reference getReference();

    Ice.ConnectionI getConnection();
    Ice.ConnectionI waitForConnection()
        throws InterruptedException, RetryException;
}

File: csharp/src/Ice/Internal/RequestHandler.cs
// Copyright (c) ZeroC, Inc.

#nullable enable

namespace Ice.Internal;

public interface CancellationHandler
{
    void asyncRequestCanceled(OutgoingAsyncBase outAsync, LocalException ex);
}

public interface RequestHandler : CancellationHandler
{
    int sendAsyncRequest(ProxyOutgoingAsyncBase outAsync);

    ConnectionI? getConnection();
}
